### PR TITLE
Add of new minecraft colours, deletion of colours that have been removed from the game

### PR DESCRIPTION
**Pull request by roimee6**
## Introduction
Minecraft has just added the colours that are linked to the materials (colours slightly modified from the classic ones) and Minecraft has removed a few other colours (I don't know since when) which are the underline and the strikethrough. 

## Tests
On this photo I realised that minecraft didn't have the underline and the strikethrough (I removed them).
image

### CODE REVIEW

File: src/utils/Terminal.php [hunk 1]
```diff
@@ -37,8 +37,6 @@ abstract class Terminal{
 	public static string $FORMAT_BOLD = "";
 	public static string $FORMAT_OBFUSCATED = "";
 	public static string $FORMAT_ITALIC = "";
-	public static string $FORMAT_UNDERLINE = "";
-	public static string $FORMAT_STRIKETHROUGH = "";
```
Comment (dktapps) [inline src/utils/Terminal.php:41]: This is BC breaking and requires a major version.

Comment (dktapps) [reply src/utils/Terminal.php:41]: In addition, consoles may have a way to represent these codes, regardless of whether they are supported by Minecraft.

File: src/utils/Terminal.php [hunk 2]
```diff
@@ -111,6 +117,16 @@ protected static function getFallbackEscapeCodes() : void{
...
+        self::$COLOR_MATERIAL_QUARTZ = $color(224);
+        self::$COLOR_MATERIAL_IRON = $color(251);
+        self::$COLOR_MATERIAL_NETHERITE = $color(234);
+        self::$COLOR_MATERIAL_REDSTONE = $color(1);
+        self::$COLOR_MATERIAL_COPPER = $color(216);
+        self::$COLOR_MATERIAL_GOLD = $color(220);
+        self::$COLOR_MATERIAL_EMERALD = $color(71);
+        self::$COLOR_MATERIAL_DIAMOND = $color(122);
+        self::$COLOR_MATERIAL_LAPIS = $color(4);
+        self::$COLOR_MATERIAL_AMETHYST = $color(171);
```
Comment (dktapps) [inline src/utils/Terminal.php:129]: What's your source for these colour codes?

File: src/utils/Terminal.php [hunk 3]
```diff
@@ -144,15 +158,25 @@ protected static function getEscapeCodes() : void{
...
-			self::$COLOR_GREEN = self::$COLOR_DARK_GREEN = $setaf(2);
-			self::$COLOR_YELLOW = self::$COLOR_GOLD = self::$COLOR_MINECOIN_GOLD = $setaf(3);
-			self::$COLOR_BLUE = self::$COLOR_DARK_BLUE = $setaf(4);
-			self::$COLOR_LIGHT_PURPLE = self::$COLOR_PURPLE = $setaf(5);
-			self::$COLOR_AQUA = self::$COLOR_DARK_AQUA = $setaf(6);
-			self::$COLOR_GRAY = self::$COLOR_WHITE = $setaf(7);
+		    self::$COLOR_MATERIAL_QUARTZ = $colors >= 256 ? $setaf(224) : $setaf(14);
+			self::$COLOR_MATERIAL_IRON = $colors >= 256 ? $setaf(251) : $setaf(7);
+			self::$COLOR_MATERIAL_NETHERITE = $colors >= 256 ? $setaf(234) : $setaf(8);
+			self::$COLOR_MATERIAL_REDSTONE = $setaf(9);
```
Comment (dktapps) [inline src/utils/Terminal.php:164]: `9` may mean different things in 16-color vs 256-color. This isn't reliable.

File: src/utils/Terminal.php [hunk 4]
```diff
@@ -144,15 +158,25 @@ protected static function getEscapeCodes() : void{
...
+			self::$COLOR_MATERIAL_COPPER = $colors >= 256 ? $setaf(216) : $setaf(3);
+			self::$COLOR_MATERIAL_GOLD = $colors >= 256 ? $setaf(220) : $setaf(3);
+			self::$COLOR_MATERIAL_EMERALD = $colors >= 256 ? $setaf(71) : $setaf(2);
+			self::$COLOR_MATERIAL_DIAMOND = $colors >= 256 ? $setaf(122) : $setaf(6);
+			self::$COLOR_MATERIAL_LAPIS = $setaf(12);
```
Comment (dktapps) [inline src/utils/Terminal.php:169]: As above.

File: src/utils/Terminal.php [hunk 5]
```diff
@@ -215,6 +237,16 @@ public static function toANSI(string $string) : string{
 				TextFormat::YELLOW => Terminal::$COLOR_YELLOW,
 				TextFormat::WHITE => Terminal::$COLOR_WHITE,
 				TextFormat::MINECOIN_GOLD => Terminal::$COLOR_MINECOIN_GOLD,
+                TextFormat::MATERIAL_QUARTZ => Terminal::$COLOR_MATERIAL_QUARTZ,
+                TextFormat::MATERIAL_IRON => seTerminallf::$COLOR_MATERIAL_IRON,
```
Comment (dktapps) [inline src/utils/Terminal.php:241]: I have no idea what's going on here, but it needs to be fixed.

File: src/utils/Terminal.php [hunk 6]
```diff
@@ -236,4 +268,4 @@ public static function write(string $line) : void{
 	public static function writeLine(string $line) : void{
 		echo self::toANSI($line) . self::$FORMAT_RESET . PHP_EOL;
 	}
-}
+}
```
Comment (dktapps) [inline src/utils/Terminal.php:271]: Useless EOF change

File: src/utils/TextFormat.php [hunk 7]
```diff
@@ -81,20 +91,26 @@ abstract class TextFormat{
...
+		self::MATERIAL_EMERALD => self::MATERIAL_EMERALD,
+		self::MATERIAL_DIAMOND => self::MATERIAL_DIAMOND,
+		self::MATERIAL_LAPIS => self::MATERIAL_LAPIS,
+		self::MATERIAL_AMETHYST => self::MATERIAL_AMETHYST,
 	];
 
 	public const OBFUSCATED = TextFormat::ESCAPE . "k";
 	public const BOLD = TextFormat::ESCAPE . "l";
-	public const STRIKETHROUGH = TextFormat::ESCAPE . "m";
-	public const UNDERLINE = TextFormat::ESCAPE . "n";
```
Comment (dktapps) [inline src/utils/TextFormat.php:90]: These can't be removed (BC breaking).

File: src/utils/TextFormat.php [hunk 8]
```diff
@@ -281,10 +289,50 @@ public static function toHTML(string $string) : string{
 					$newString .= "<span style=color:#FFF>";
 					++$tokens;
 					break;
-				case TextFormat::MINECOIN_GOLD:
+				case TextFormat::MINECOIN_GOLD :
```
Comment (dktapps) [inline src/utils/TextFormat.php:292]: Formatting issue here.

File: src/utils/TextFormat.php [hunk 9]
```diff
@@ -281,10 +289,50 @@ public static function toHTML(string $string) : string{
...
+				case TextFormat::MATERIAL_DIAMOND:
+					$newString .= "<span style=color:#2cb9a8>";
+					++$tokens;
+					break;
+				case TextFormat::MATERIAL_LAPIS:
+					$newString .= "<span style=color:#20487a>";
+					++$tokens;
+					break;
+				case TextFormat::MATERIAL_AMETHYST:
+					$newString .= "<span style=color:#9a5cc5>";
```
Comment (dktapps) [inline src/utils/TextFormat.php:333]: Source for these hex codes?

File: src/utils/TextFormat.php [hunk 10]
```diff
@@ -295,4 +343,4 @@ public static function toHTML(string $string) : string{
 
 		return $newString;
 	}
-}
+}
```
Comment (dktapps) [inline src/utils/TextFormat.php:346]: Useless EOF change

**Review by dktapps — CHANGES_REQUESTED** [review]
Again, please do **not** use the GitHub Web Editor or use it to upload files. Install git from https://git-scm.com and use it to commit your changes.

**Comment by dktapps** [comment]
Bump. Are you planning to complete this PR?

**Comment by roimee6** [comment]
New pr

**Comment by dktapps** [comment]
You didn't have to do that... it's entirely possible to modify an existing PR

**Pull request closed without merging**